File: spec/python/test_nested_same_name2.py
import unittest

from nested_same_name2 import NestedSameName2

class TestNestedSameName2(unittest.TestCase):
    def test_nested_same_name2(self):
        r = NestedSameName2.from_file("src/nested_same_name2.bin")

        self.assertEquals(r.version, 0x42)
        self.assertEquals(r.main_data.main_size, 2)
        self.assertEquals(r.main_data.foo.data1, "\x11\x11\x11\x11")
        self.assertEquals(r.dummy.dummy_size, 3)
        self.assertEquals(r.dummy.foo.data2, "\x22\x22\x22\x22\x22\x22")

File: spec/lua/test_nested_same_name2.lua
local luaunit = require("luaunit")

require("nested_same_name2")

TestNestedSameName2 = {}

function TestNestedSameName2:test_nested_same_name2()
    local r = NestedSameName2:from_file("src/nested_same_name2.bin")

    luaunit.assertEquals(r.version, 0x42)
    luaunit.assertEquals(r.main_data.main_size, 2)
    luaunit.assertEquals(r.main_data.foo.data1, "\x11\x11\x11\x11")
    luaunit.assertEquals(r.dummy.dummy_size, 3)
    luaunit.assertEquals(r.dummy.foo.data2, "\x22\x22\x22\x22\x22\x22")
end
